File: backend/app/schemas/resume_checker.py
from typing import Optional, List, Dict, Any
from pydantic import BaseModel


class ResumeCheckerRequest(BaseModel):
    resume_text: Optional[str] = None
    resume_source: Optional[str] = "master_profile"  # "master_profile" | "uploaded_text" | "tailored_resume"
    jd_text: Optional[str] = None
    jd_url: Optional[str] = None
    job_id: Optional[int] = None


class ExtractedCategorySchema(BaseModel):
    technical_skills: List[str] = []
    tools: List[str] = []
    soft_skills: List[str] = []
    certifications: List[str] = []
    role_titles: List[str] = []


class ResumeCheckerResponse(BaseModel):
    overall_ats_score: int
    keyword_match_score: int
    formatting_score: int
    completeness_score: int
    matched_keywords: List[str] = []
    missing_keywords: List[str] = []
    resume_keywords: ExtractedCategorySchema = ExtractedCategorySchema()
    jd_keywords: ExtractedCategorySchema = ExtractedCategorySchema()
    recommendations: List[str] = []
    resume_preview_text: Optional[str] = None
    jd_preview_text: Optional[str] = None


class SaveFingerprintRequest(BaseModel):
    keywords: List[str] = []

File: frontend/src/types/resume_checker.ts
export interface ExtractedCategory {
  technical_skills: string[];
  tools: string[];
  soft_skills: string[];
  certifications: string[];
  role_titles: string[];
}

export interface ResumeCheckerResponse {
  overall_ats_score: number;
  keyword_match_score: number;
  formatting_score: number;
  completeness_score: number;
  matched_keywords: string[];
  missing_keywords: string[];
  resume_keywords: ExtractedCategory;
  jd_keywords: ExtractedCategory;
  recommendations: string[];
  resume_preview_text?: string;
  jd_preview_text?: string;
}

export interface ResumeCheckerRequest {
  resume_text?: string;
  resume_source?: 'master_profile' | 'uploaded_text' | 'tailored_resume';
  jd_text?: string;
  jd_url?: string;
  job_id?: number;
}
